File: daily/rust/27_remove_element.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut i = 0;
        let mut size = nums.len();
        while i < size {
            if nums[i] == val {
                nums[i] = nums[size - 1];
                size -= 1;
            } else {
                i += 1;
            }
        }
        size as i32
    }
}

File: daily/cpp/27_remove_element.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int removeElement(vector<int>& nums, int val) {
    int size = nums.size();
    int i = 0;
    while (i < size) {
      if (nums[i] == val) {
        nums[i] = nums[size - 1];
        size--;
      } else {
        i++;
      }
    }
    return size;
  }
  /** A typical double pointer */
};